src/sentry/api/endpoints/grouping_level_new_issues.py

```diff
@@ -0,0 +1,212 @@
+    def get(self, request, id: str, group: Group):
+        """
+        Retrieve information about a particular grouping level, including a
+        list of issues it would create.
+
+        ```
+        GET /api/0/issues/<group_id>/grouping/levels/<level_id>/new-issues/
+
+        [
+            {"hash": "...", "latestEvent": ..., "eventCount": 132},
+            ...
+        ]
+        ```
+
+        Available level IDs can be fetched from `GroupingLevelsEndpoint`.
+
+        Each row/array item corresponds to one *new issue* that selecting this
+        level would create in place of the *affected issues*. The array items
+        are not groups, but groups that will be created, therefore a lot of
+        information normally available for groups is missing.
+
+        - `latestEvent`: a sample event in the same format returned by the
+          event details endpoint(s).
+
+        - `hash`: The grouping hash, probably insignificant to the user but can
+          be shown for diagnostic purposes.
+
+        - `eventCount`: How many events this issue would contain. Note that
+          like with any other event count, this number can change all the time
+          because events keep coming in.
+
+        The "would-be issues" are returned in-order such that the most recently
+        seen "issue" is at the top, i.e. it is sorted in descending order of
+        `latestEvent.dateCreated`.
+
+        The *affected issue* (=to-be-deleted issue) is often just the current one,
+        however if the previewed grouping level is reduced, this endpoint can
+        return a list of entries which together have more events than the
+        current issue (meaning issues will be merged together).
+
+        In the future there will be an endpoint that allows you to fetch the
+        list of affected issues. For now the UI should simply show a warning if
+        the level is decreased (and possibly only if the summed up events of
+        the new issues are more than what the current issue has).
+        """
+
+        if not features.has(
+            "organizations:grouping-tree-ui", group.project.organization, actor=request.user
+        ):
+            return self.respond(
+                {"error": "This project does not have the grouping tree feature"},
+                status=403,
+            )
+
+        def data_fn(offset=None, limit=None):
+            return _query_snuba(group, id, offset=offset, limit=limit)
+
+        def on_results(results):
+            return _process_snuba_results(results, group, request.user)
+
+        return self.paginate(
+            request=request,
+            on_results=on_results,
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+        )
+
+
+def _get_hash_for_parent_level(group: Group, id: int, levels_overview: LevelsOverview) -> str:
+    # If this is violated, there cannot be a 1:1 mapping between level and hash.
+    assert 0 <= id < levels_overview.current_level
```

Comment:
Can I provoke a 500 response by passing `-1` as id?

Reply:
I know this is weird but I am doing this because I want to monitor which levels the UI passes back into here, and be alerted too because I probably returned a wrong level in the other endpoint.

Same with all the unguarded `int()` accesses.

If the API gets more stable and we get clients that are outside of our control we can reevaluate. But right now I prefer to have an alert in the backend Sentry project rather than getting a random UI UnhandledPromiseRejection alert that I can't decipher.

Reply:
Makes sense.